internal/controllers/topology/cluster/cluster_controller.go

```diff
@@ -240,6 +263,25 @@ func (r *Reconciler) setupDynamicWatches(ctx context.Context, s *scope.Scope) er
 	return nil
 }
 
+func (r *Reconciler) callBeforeClusterCreateHook(ctx context.Context, s *scope.Scope) (reconcile.Result, error) {
```

Comment:
This logic is separated into a separate block instead of inline it in the `reconcile` block because it simplifies unit testing this piece of code.